components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -198,15 +200,18 @@ export const HierarchicalViewMixin = superclass => class extends superclass {
 	}
 
 	getRootView() {
-		if (!this.childView) {
+		if (this.rootView || !this._childView) {
 			return this;
 		}
 		const rootView = findComposedAncestor(
 			this.parentNode,
 			(node) => {
-				return node.hierarchicalView && !node.childView;
+				return node.rootView || (node.hierarchicalView && !node._childView);
 			}
 		);
+		if (rootView) {
+			rootView.rootView = true;
```

Comment:
Curious about thoughts on this - if the rootView were to change then I believe this could cause problems. For example:
```
<d2l-menu root-view id="menu-1">
<d2l-menu child-view id="menu-2">
```

but then we add another view above:
```
<d2l-menu id="new-menu-1">
<d2l-menu root-view id="menu-1">
<d2l-menu child-view id="menu-2">
```

Though I believe this would also be a problem with the current childView logic, as if something that's a child-view becomes the root-view, it would still have the child-view attribute on it, so perhaps this is a case we don't currently support.

Reply:
There is the method `__isChildView()` which is misleading. It actually updates the `childView` property, and is called from `connectedCallback()` and `firstUpdated()`. 

Kinda looks like `firstUpdated` call handles the initial case. Perhaps the `connectedCallback` call handles a scenario where the hierarchical-view is moved in the DOM (a scenario which could happen in theory, but unlikely in practice). But I think we've also had timing issues before.

I'm wondering if we should similarly initialize `rootView`:

* If the consumer has specified `root-view`, then `rootView = true`
* If the consumer has not specified `root-view` then we walk up the DOM...
  * If we don't find another hierarchical-view, then `rootView = true`
  * If we find another hierarchical-view then `rootView = false`

So if a view is added above, it depends on how that's done. If new DOM is being constructed to insert the element above in the DOM, then `firstUpdated` should handle that case. If the hierarchical-view is being moved in the DOM (ex. via `removeChild` and then `appendChild`), the `connectedCallback` should handle that case. In either scenario, I _think_ we want to re-initialize `rootView`. It's up to the new parent to force `rootView = true` if it wants, and if it doesn't, `rootView` is re-initialized using the DOM walk.

With `rootView` always initialized, maybe we can update `d2l-menu` to use it instead of `_childView`.

Reply:
Cool I think that makes sense to me. I'll update that. Thanks!

Reply:
I've made the suggested changes. There are other places in hierarchical mixin that we could switch `_childView` for something like `!rootView` but I'm not sure how far we want to go down that road as part of this.

Reply:
Yeah not sure. It'd be nice to clean-up `childView` entirely, but I was sort of thinking that changing those internals would be more than we'd want to do right now, mostly just to minimize risk.

Reply:
(but if there's more that you want to include, go for it!)

Reply:
I'm good with a safer approach here